test/filter.jl

```diff
@@ -108,4 +108,17 @@ h5open(fn, "r") do f
     @test f["filtshufdef"][] == data
 end
 
+# Filter Pipeline test for UnknownFilter
+FILTERS_backup = copy(HDF5.Filters.FILTERS)
+empty!(HDF5.Filters.FILTERS)
+h5open(fn, "w") do f
+    data = collect(1:128)
+    filter = UnknownFilter(H5Z_FILTER_LZ4, 0, Cuint[0, 2, 4, 6, 8, 10], "Unknown LZ4", 0)
```

Comment:
Need to manually import the `UnknownFilter` type, since it technically shouldn't be exported: https://github.com/JuliaIO/HDF5.jl/pull/906